libs/ui/search/src/lib/ui-search.module.ts

```diff
@@ -18,5 +20,17 @@ export * from './search/search.component';
   ],
   exports: [TsSearchComponent],
   declarations: [TsSearchComponent],
+  providers: [
+    { provide: TS_SEARCH_DEFAULT_OPTIONS, useValue: defaultOptions }
+  ]
 })
-export class TsSearchModule {}
+export class TsSearchModule {
+  static forRoot(options: TsSearchDefaultOptions): ModuleWithProviders<TsSearchModule> {
+    return {
+      ngModule: TsSearchModule,
+      providers: [
+        { provide: TS_SEARCH_DEFAULT_OPTIONS, useValue: options }
```

Comment:
looks like `useValue`  should be 
```
useValue: {
  ...defaultOptions,
  ...options,
}
```
to prevent errors, if user doesn't provide some field

Reply:
fixed